src/Polly.Core/Registry/ResiliencePipelineRegistry.TResult.cs

```diff
@@ -65,9 +61,5 @@ public ResiliencePipeline<TResult> GetOrAdd(TKey key, Action<ResiliencePipelineB
         }
 
         public bool TryAddBuilder(TKey key, Action<ResiliencePipelineBuilder<TResult>, ConfigureBuilderContext<TKey>> configure) => _builders.TryAdd(key, configure);
-
-        public bool RemoveBuilder(TKey key) => _builders.TryRemove(key, out _);
-
-        public void Clear() => _strategies.Clear();
```

Comment:
I think we should maybe keep clear at least. Relying on reloading to do that for you feels a bit unintuitive to me.

Reply:
The follow-up of this PR is the dispose support for the registry shich is something that's missing right now. It essentially frees resources held by individual resilience pipelines (`RateLimiter`, `CircuitBreaker`).

Now the problem is:

``` csharp
var registry = new ResiliencePipelineRegistry<string>();
var pipeline= registry.GetPipeline("my-pipeline");

// this call also disposes all resources held by "my-pipeline"
registry.ClearPipelines();

// ObjectDisposedException
pipeline.Execute(()=> {  });
```

For this reason I thought it might be better to just wait and see. If someone really needs it we can add it, but the consumer should be aware of side-effects.

Reply:
The follow up for this PR is the disposal of resources that are held by the resilience pipelines.

Think R